src/main/resources/services/dashboard/dashboard.es6

```diff
@@ -1,172 +1,226 @@
+      return {
+        dataset,
+        deleted: content.delete({
+          key: dataset._id
         })
-        message = `Successfully deleted ${datasetInfo.length} of ${datasets.length} datasets`
-      } else {
-        success = false
-        message = `No dataset found for dataquery with id ${params.id}`
-        status = 404
       }
     })
-  } else {
-    success = false
-    message = `Missing parameter "id"`
-    status = 400
+  })
+
+  if(deleteResult.length === 0) {
+    logDataQueryEvent(req.params.id, {message: Events.DELETE_FAILED, deleteResult})
+    return successResponse(deleteResult, undefined)
   }
 
+  logDataQueryEvent(req.params.id, {message:Events.DELETE_OK, deleteResult })
+
+  const publishResult = content.publish({
+    keys: deleteResult.map((result) => result.dataset._id),
+    sourceBranch: 'draft',
+    targetBranch: 'master'
+  })
+
+  publishResult.deletedContents.forEach((id) => {
+    logDataQueryEvent(id, {message:Events.DELETE_OK_PUBLISHED, deleteResult })
+  })
+
+  publishResult.failedContents.forEach((id) => {
+    logDataQueryEvent(id, {message:Events.DELETE_FAILED_PUBLISHED, deleteResult })
+  })
+
+  const updateResult = deleteResult.map( (result) => { // refreshResult
+    const queryLogNode = getNode(EVENT_LOG_REPO, EVENT_LOG_BRANCH,`/queries/${result.dataset.data.dataquery}`)
+    return {
+      id: result.dataset.data.dataquery,
+      message: queryLogNode.data.modifiedResult ? i18n.localize({key: queryLogNode.data.modifiedResult}): undefined,
+      status: queryLogNode.data.modifiedResult,
+      dataset: {
+        newDatasetData: true,
+        modified: '',
+        modifiedReadable: ''
+      },
+      logData: {
+        ...queryLogNode.data,
+        message: i18n.localize({key: queryLogNode.data.modifiedResult}),
+        modified: dateToFormat(queryLogNode.modified),
+        modifiedReadable: dateToReadable(queryLogNode.modified)
+      }
+    }
+  })
+
+  return successResponse(updateResult, publishResult)
+}
+
+/**
+ *
+ * @param {object} updateResult
+ * @param {object} publishResult
+ * @return {{body: {publishResult: *, updates: *, message: *}, contentType: string, status: number}}
+ */
+function successResponse(updateResult, publishResult) {
   return {
     body: {
-      success,
-      message,
-      updates: datasetInfo
+      updates: updateResult,
+      message: createMessage(updateResult),
+      publishResult
     },
     contentType: 'application/json',
-    status
+    status: 200
   }
 }
 
+/**
+ *
+ * @return {{body: {success: boolean, message: string}, contentType: string, status: number}}
+ */
+function missingParameterResponse() {
+  return {
+    body: {
+      success: false,
+      message: `Missing parameter "id"`
+    },
+    contentType: 'application/json',
+    status: 400
+  }
+}
+
+/**
+ *
+ * @param {object }updateResult
+ * @return {string|string|"NOT_TRANSLATED"}
+ */
+function createMessage(updateResult) {
+  if ( updateResult.length === 0) {
```

Comment:
`Array.isArray(updateResult)  && (updateResult.length ...)` might be a safer bet here ?